src/NuGet.Core/NuGet.PackageManagement/Resolution/ResolverGather.cs

```diff
@@ -560,10 +560,15 @@ private void QueueWork(IReadOnlyList<SourceResource> sources, PackageIdentity pa
             {
                 foreach (SourceResource source in sources)
                 {
-                    if (_areNamespacesEnabled && configuredPackageSources != null && !configuredPackageSources.Contains(source.Source.PackageSource.Name, StringComparer.CurrentCultureIgnoreCase))
+                    if (_areNamespacesEnabled)
                     {
-                        // This package's id prefix is not defined in current package source, let's skip.
-                        continue;
+                        if (configuredPackageSources == null ||
+                            configuredPackageSources.Count == 0 ||
+                            !configuredPackageSources.Contains(source.Source.PackageSource.Name, StringComparer.CurrentCultureIgnoreCase))
```

Comment:
I am curious why `StringComparer.CurrentCultureIgnoreCase`?

Reply:
I'm already doing this for `PackageDownloader.cs`, it was recommended during previous PR review. 
Here we're case in-sensitive.